twinkle.js

```diff
@@ -262,11 +262,24 @@ Twinkle.addPortlet = function(navigation, id, text, type, nextnodeid) {
 		}
 		return null;
 	}
+
 	if (type === 'menu') {
 		// In order to get mw.util.addPortlet to generate a dropdown menu in vector and vector-2022, the nextnodeid must be p-cactions. Any other nextnodeid will generate a non-dropdown portlet instead.
 		nextnodeid = 'p-cactions';
 	}
-	return mw.util.addPortlet(id, text, '#' + nextnodeid);
+
+	let portlet = mw.util.addPortlet(id, text, '#' + nextnodeid);
+
+	if (mw.config.get('skin') === 'vector') {
+		$('#p-twinkle').insertAfter('#p-cactions');
+	} else if (mw.config.get('skin') === 'vector-2022') {
+		$('#p-twinkle-dropdown').insertAfter('.vector-page-tools-landmark');
```

Comment:
`vector-page-tools-landmark` is not stable so may change at any point. If that's the case I think jQuery will quietly ignore this code but not move it. With that in mind would it make sense to check and log that situation to aid debuggin?

```
if ( !document.querySelector( '.vector-page-tools-landmark' ) ) {
  mw.log.warn('Unexpected change in DOM' );
}